### PR TITLE
distro/rhel90: enable edge-simplified-installer image type

### PR DESCRIPTION
**Pull request by achilleas-k**
This reverts commit 226ada67f93cdfda73b3cae69d45eaf4fb281d6f.

This PR replaces #2010 and (I think) #1744.

### CODE REVIEW

**Comment by achilleas-k** [comment]
Changed RHEL 9.0 `edge-simplified-installer` to use `coreos-installer-dracut`, based on RHEL 8.6 changes by @runcom in PR #1752.

**Comment by henrywang** [comment]
@achilleas-k I found the error `DNF error occured: MarkingErrors: Error occurred when marking packages for installation: Problems in request:\nmissing packages: coreos-installer, coreos-installer-dracut`. Looks the [cdn](https://github.com/osbuild/osbuild-composer/blob/6757916c549ca7f43f17b18dc9f2e1d05d3d8b5a/repositories/rhel-90.json#L60) repo does not have `coreos-installer` included, but nightly has.

**Comment by runcom** [comment]
Shouldn’t we just update the osbuild mirror? Or use nightly (to also unblock my other PRs for 8.6

**Comment by jrusz** [comment]
It's actually using rpmrepo snapshot for the repositories and I guess that doesn't have the package in it yet... You can see it if you look for `composer-cli sources list` in the log...
```
++ sudo composer-cli sources list
+ for SOURCE in $(sudo composer-cli sources list)
+ sudo composer-cli sources info appstream
check_gpg = false
check_ssl = true
id = "appstream"
name = "appstream"
rhsm = false
system = true
type = "yum-baseurl"
url = "https://rpmrepo.osbuild.org/v2/mirror/rhvpn/el9/el9-x86_64-appstream-n9.0-20211105/"
+ for SOURCE in $(sudo composer-cli sources list)
+ sudo composer-cli sources info baseos
check_gpg = false
check_ssl = true
id = "baseos"
name = "baseos"
rhsm = false
system = true
type = "yum-baseurl"
url = "https://rpmrepo.osbuild.org/v2/mirror/rhvpn/el9/el9-x86_64-baseos-n9.0-20211105/"
+ for SOURCE in $(sudo composer-cli sources list)
+ sudo composer-cli sources info rt
check_gpg = false
check_ssl = true
id = "rt"
name = "rt"
rhsm = false
system = true
type = "yum-baseurl"
url = "https://rpmrepo.osbuild.org/v2/mirror/rhvpn/el9/el9-x86_64-rt-n9.0-20211105/"
```
This happens during sourcing  `tools/provision.sh` https://github.com/osbuild/osbuild-composer/blob/main/tools/provision.sh#L75

**Review by diaasami — DISMISSED** [review]

**Comment by achilleas-k** [comment]
I asked Tomas to create a new repo snapshot and I'll update the repo links once it's done.

File: internal/distro/rhel90/pipelines.go [hunk 1]
```diff
@@ -1199,7 +1199,7 @@ func simplifiedInstallerTreePipeline(repos []rpmmd.RepoConfig, packages []rpmmd.
 	p.AddStage(osbuild.NewBuildstampStage(buildStampStageOptions(arch, product, osVersion, variant)))
 	p.AddStage(osbuild.NewLocaleStage(&osbuild.LocaleStageOptions{Language: "en_US.UTF-8"}))
 	p.AddStage(osbuild.NewSystemdStage(systemdStageOptions([]string{"coreos-installer"}, nil, nil, "")))
```
Comment (runcom) [inline internal/distro/rhel90/pipelines.go:1201]: drop this line as well (from https://github.com/osbuild/osbuild-composer/pull/1752/files#diff-96e355e931e7a80acc491324230af1bd2af9cccb72609a76c246d923e02a773eL1202)

**Comment by runcom** [comment]
> I asked Tomas to create a new repo snapshot and I'll update the repo links once it's done.

we need a new snapshot for 8.6 as well for the 8.6 PR version of this too as that's failing too cause there's no coreos-installer-dracut in rpmrepo.osbuild https://github.com/osbuild/osbuild-composer/pull/1752

**Comment by achilleas-k** [comment]
Updated the test repositories and rebased on main.

File: internal/distro/rhel90/stage_options.go [hunk 2]
```diff
@@ -325,7 +325,7 @@ func grubISOStageOptions(installDevice, kernelVer, arch, vendor, product, osVers
 				"console=ttyS0",
 				"systemd.log_target=console",
 				"systemd.journald.forward_to_console=1",
-				"edge.liveiso=" + isolabel,
+				"coreos.insta.isoroot=" + isolabel,
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply internal/distro/rhel90/stage_options.go]: Thank you so much.  I would have had a hard time finding this probably.

File: internal/distro/rhel90/pipelines.go (hunk 1 above)
Comment (achilleas-k) [reply internal/distro/rhel90/pipelines.go]: Good catch! :+1:

File: internal/distro/rhel90/pipelines.go [hunk 3]
```diff
@@ -1198,8 +1198,7 @@ func simplifiedInstallerTreePipeline(repos []rpmmd.RepoConfig, packages []rpmmd.
 	p.AddStage(osbuild.NewRPMStage(rpmStageOptions(repos), rpmStageInputs(packages)))
 	p.AddStage(osbuild.NewBuildstampStage(buildStampStageOptions(arch, product, osVersion, variant)))
 	p.AddStage(osbuild.NewLocaleStage(&osbuild.LocaleStageOptions{Language: "en_US.UTF-8"}))
-	p.AddStage(osbuild.NewSystemdStage(systemdStageOptions([]string{"coreos-installer"}, nil, nil, "")))
-	p.AddStage(osbuild.NewDracutStage(dracutStageOptions(kernelVer, arch, []string{"rdcore"})))
+	p.AddStage(osbuild.NewDracutStage(dracutStageOptions(kernelVer, arch, []string{"coreos-installer"})))
```
Comment (runcom) [inline internal/distro/rhel90/pipelines.go:1201]: looks like the test in rhel9 can't find this module even if `coreos-installer-dracut` is installed?

Comment (runcom) [reply internal/distro/rhel90/pipelines.go:1201]: nevermind, rebuilding the package cause of a bug in how it installs dracut modules on the system

Comment (runcom) [reply internal/distro/rhel90/pipelines.go:1201]: we'd need a new rpmrepo snapshot too

**Comment by runcom** [comment]
I think we can also get https://github.com/osbuild/osbuild-composer/pull/2045 before this so we don't introduce rdma-core - we don't need that in the simplified installer at all

**Comment by achilleas-k** [comment]
This requires `coreos-installer-dracut` version `0.10.1-4`, which should appear in the nightly composes next week.

**Comment by runcom** [comment]
> This requires coreos-installer-dracut version 0.10.1-4, which should appear in the nightly composes next week.

it's now in the nightly repos - we need another snapshot and we should be good to go :+1:

File: internal/distro/rhel90/package_sets.go [hunk 4]
```diff
@@ -668,6 +668,7 @@ func edgeSimplifiedInstallerPackageSet(t *imageType) rpmmd.PackageSet {
 			"policycoreutils",
 			"policycoreutils-python-utils",
 			"procps-ng",
+			"rdma-core",
```
Comment (runcom) [inline internal/distro/rhel90/package_sets.go:671]: this can be dropped now that #2045 merged

Comment (achilleas-k) [reply internal/distro/rhel90/package_sets.go]: Done.  Thanks!

**Comment by runcom** [comment]
@achilleas-k can we request/do another snapshot to make test pass here? last nightlies contain the package we need with the correct version

**Comment by achilleas-k** [comment]
Pointed test repos to today's snapshot

**Comment by runcom** [comment]
Looks like some resource constraints in CI

**Comment by runcom** [comment]
Google api error 😅

**Comment by achilleas-k** [comment]
Also needed a rebase on main

**Comment by runcom** [comment]
looks like mkfs.fat in osbuild has some issues https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1816235011#L3130
apart from that, I think coreos-installer-dracut is now properly working in RHEL9 :)

**Comment by achilleas-k** [comment]
> looks like mkfs.fat in osbuild has some issues https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1816235011#L3130 apart from that, I think coreos-installer-dracut is now properly working in RHEL9 :)

Posted in chat but should've commented here:
Empty volid is invalid in `mkfs.fat` (I guess it was always invalid, but they fixed the validation in the latest release):
https://github.com/dosfstools/dosfstools/commit/07e63ac7efa6fc94fb9e394b6ec8d9a5ebbe79e9

You can omit the flag entirely in the command, but in osbuild it's required in the stage options:
https://github.com/osbuild/osbuild/blob/2e9ffaf517e1d717be2f588baea98794b0f21ae3/stages/org.osbuild.mkfs.fat#L32

so not specifying it means it gets passed as empty string, which breaks now.

**Comment by achilleas-k** [comment]
`coreos-installer` needs an another update/fix for this.  It removes the `initrd-release` file in the initramfs, which seems to upset systemd now.

**Comment by runcom** [comment]
> `coreos-installer` needs an another update/fix for this. It removes the `initrd-release` file in the initramfs, which seems to upset systemd now.

the fix landed in nightlies, we can snapshot that and retest here

**Comment by henrywang** [comment]
@achilleas-k Tests on RHEL 9.0 failed at `DNF error occured: RepoError: Error occurred when setting up repo: Failed to download metadata for repo '0': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried`
I think the nightly compose repo URL should be `http://download.devel.redhat.com/rhel-9/nightly/RHEL-9/latest-RHEL-9.0/compose/BaseOS/x86_64/os`. The repo URL needs `/os`.

**Comment by achilleas-k** [comment]
Thanks @henrywang.  I neglected this PR **again**.  Changing now.

**Comment by runcom** [comment]
LGTM :)

**Review by lavocatt — APPROVED** [review]

**Pull request merged**